src/isd/batch.py

```diff
@@ -0,0 +1,87 @@
+import gzip
+import json
+from io import BytesIO
+from pathlib import Path
+from dataclasses import dataclass
+from typing import List, Union, Optional, Dict, Any, Iterator
+import datetime as dt
+
+from isd.record import Record
+
+import pandas as pd
```

Comment:
My personal style preference is to _not_ abbreviate package/module names.

```suggestion
import datetime

from isd.record import Record

import pandas
```